src/widget/wwidget.cpp

```diff
@@ -37,6 +38,31 @@ WWidget::~WWidget() {
     delete m_pTouchShift;
 }
 
+double WWidget::getBackgroundColorRgba() const {
+    if (m_backgroundColorRgba >= 0) {
+        return m_backgroundColorRgba;
```

Comment:
It's more readable to make the guard clause the exceptional case, it better communicates which is the primary branch (unindented) and which is the error case (the indented part)

```
if (m_backgroundColorRgba < 0) {
  return -1;
}
return m_backgroundColorRgba;
```